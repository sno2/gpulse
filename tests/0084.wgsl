// Array where:
//   - alignment is 4 = AlignOf(f32)
//   - element stride is 4 = roundUp(AlignOf(f32),SizeOf(f32)) = roundUp(4,4)
//   - size is 32 = stride * number_of_elements = 4 * 8
var small_stride: array<f32, 8>;

// Array where:
//   - alignment is 16 = AlignOf(vec3<f32>) = 16
//   - element stride is 16 = roundUp(AlignOf(vec3<f32>), SizeOf(vec3<f32>))
//                          = roundUp(16,12)
//   - size is 128 = stride * number_of_elements = 16 * 8
var bigger_stride: array<vec3<f32>, 8>;
