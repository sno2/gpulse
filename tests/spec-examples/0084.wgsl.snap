ast.Node{ .var_decl = ast.VarDecl{ .name = ast.Span{ .start = 191, .end = 203 }, .access_mode = ast.AccessMode.read_write, .addr_space = null, .typ = ast.Node{ .template = ast.Template{ .name = ast.Span{ .start = 205, .end = 210 }, .args = { ast.Node{ .identifier = ast.Span{ .start = 211, .end = 214 } }, ast.Node{ .number_literal = ast.NumberLiteral{ .span = ast.Span{ .start = 216, .end = 217 }, .kind = ast.NumberLiteral.NumberKind.abstract_int } } } } }, .value = null } }
ast.Node{ .var_decl = ast.VarDecl{ .name = ast.Span{ .start = 470, .end = 483 }, .access_mode = ast.AccessMode.read_write, .addr_space = null, .typ = ast.Node{ .template = ast.Template{ .name = ast.Span{ .start = 485, .end = 490 }, .args = { ast.Node{ .template = ast.Template{ .name = ast.Span{ .start = 491, .end = 495 }, .args = { ast.Node{ .identifier = ast.Span{ .start = 496, .end = 499 } } } } }, ast.Node{ .number_literal = ast.NumberLiteral{ .span = ast.Span{ .start = 502, .end = 503 }, .kind = ast.NumberLiteral.NumberKind.abstract_int } } } } }, .value = null } }
